src/plugins/kibana_react/public/table_list_view/table_list_view.tsx

```diff
@@ -65,7 +65,14 @@ export interface TableListViewProps<V> {
   tableCaption: string;
   searchFilters?: SearchFilterConfig[];
   theme: ThemeServiceStart;
-  application: ApplicationStart;
+  application:
+    | ApplicationStart
+    | {
+        getUrlForApp: ApplicationStart['getUrlForApp'];
+        capabilities: {
+          advancedSettings: ApplicationStart['capabilities']['advancedSettings'];
+        };
+      };
```

Comment:
Added this to narrow down the application service to only require the functionality that is actually used by the table list view - that's so that the Dashboard app can support a partial ApplicationStart rather than the entire service.